plugins/heroku/src/buildHerokuReviewApp.ts

```diff
@@ -1,25 +1,23 @@
-import heroku from './herokuClient'
+import heroku, { extractHerokuError } from './herokuClient'
 import type { HerokuApiResGetReview, HerokuApiResPost } from 'heroku-client'
 import { Logger } from 'winston'
 import { ToolKitError } from '@dotcom-tool-kit/error'
 import { repeatedCheckForSuccessStatus } from './repeatedCheckForSuccessStatus'
 import { getRepoDetails } from './githubApi'
 
-async function buildHerokuReviewApp(
-	logger: Logger,
-	pipelineId: string
-	): Promise<string> {
-
+async function buildHerokuReviewApp(logger: Logger, pipelineId: string): Promise<string> {
   const { branch, repo, source_blob } = await getRepoDetails(logger)
 
   logger.info(`creating review app for ${branch} branch on ${repo}...`)
-  const reviewAppBuild: HerokuApiResPost = await heroku.post(`/review-apps`, {
-    body: {
-      branch: branch,
-      pipeline: pipelineId,
-      source_blob
-    }
-  })
+  const reviewAppBuild = await heroku
+    .post<HerokuApiResPost>(`/review-apps`, {
```

Comment:
**praise**: nice work adding these generics to all the calls, that'll definitely help maintainability of this plugin

Reply:
no need to praise me, tsc fails if you don't put them here! 😅 previously it could infer the type parameter based on the type of the variable we're assigning the return value to, but now that we weave the result through a `catch` statement too the type inference fails, even though `extractHerokuError` is diverging and therefore doesn't change the return value